Objects.Inheritance.py

```diff
@@ -0,0 +1,136 @@
+'''
+Creation of a class "figure" with properties: coordinates of 4 points. The class contains methods
+for checking the existence of the created quadrilateral, calculation and output of information
+about a figure: lengths of sides, diagonals, perimeter, area of a figure.
+The class has a check of the figure for convexity.
+Added derived class "parallelogram". The class provides a check whether the figure is a parallelogram.
+
+Below is a commented out program that analyzes quadrilaterals to find the maximum area among objects.
+'''
+import math
+
+
+class Figure:
+    def __init__(self, x1, y1, x2, y2, x3, y3, x4, y4):
+        if x1 == x2 == x3 == x4 or y1 == y2 == y3 == y4:
```

Comment:
This check could be enhanced by checking the angle coefficient of sides, instead of comparing `x` or `y` coordinates. In this case, you'll cover situations when all of the points lay on one diagonal line.

Reply:
done